.github/workflows/publish.yml

```diff
@@ -40,9 +49,39 @@ env:
   CACHE: true
 
 jobs:
+  semver_check:
+    name: Check Semver
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          cargo-cache-key: cargo-semver-checks
+          toolchain: test
+          components: edit, release, semver-checks
+
+      - name: Set Version
+        run: |
+          if [ "${{ inputs.level }}" == "version" ]; then
+            LEVEL=${{ inputs.version }}
+          else
+            LEVEL="--bump ${{ inputs.level }}"
+          fi
+
+          cargo set-version --manifest-path ${{ inputs.crate }}/Cargo.toml $LEVEL
```

Comment:
Actually, sorry I only just thought about this, but `cargo release` can actually do this too if you run:

```
cargo release <LEVEL> --no-tag --no-publish --no-push --no-confirm --execute
```

And that way we don't need to use a new tool. What do you think?

Reply:
Ah, I see, yeah that could a be a good idea.